components/ResourceCodeEditor.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import CodeMirror from "@uiw/react-codemirror";
+import { useState } from "react";
+import { json, jsonParseLinter } from "@codemirror/lang-json";
+import { linter } from "@codemirror/lint";
+import { Button } from "@mantine/core";
+
+const jsonLinter = jsonParseLinter();
+
+/**
+ * interface for the props that a ResourceCodeEditor component takes in.
+ * initialValue is what the code editor will contain initially
+ * onClickFunction is the function that should be executed when the ResourceCodeEditor's submit button is clicked.
+ */
+export interface ResourceCodeEditorProps {
+  initialValue: string;
+  onClickFunction: (submittedVal: string) => void;
+  buttonName: string;
+}
+
+/**
+ * ResourceCodeEditor is a component for rendering a Code Editor with a submit Button component that executes the function specified in the props.
+ * The function called when the submit button is clicked is passed the Code Editor's current contents
+ * @param props see interface ResourceCodeEditorProps
+ * @returns React node composed of a CodeMirror component and a submit button that executes the function passed via props
+ */
+const ResourceCodeEditor = (props: ResourceCodeEditorProps) => {
+  const [currentValue, setCurrentValue] = useState(props.initialValue);
+  const [hasLintError, setHasLinterError] = useState(true);
+
+  return (
+    <div>
+      <CodeMirror
+        data-testid="resourceCodeEditor"
+        value={currentValue}
+        height="500px"
+        extensions={[json(), linter(jsonLinter)]}
+        onUpdate={(v) => {
+          const diagnosticMessages = jsonLinter(v.view).map((d) => d.message);
+
+          if (diagnosticMessages.length === 0) {
+            setHasLinterError(false);
+          } else {
+            setHasLinterError(true);
+          }
+          setCurrentValue(v.state.toJSON().doc);
+        }}
+      />
+      <br /> <br />
+      <div style={{ textAlign: "center" }}>
+        <Button
```

Comment:
I'm thinking about the re-usability of this component down the line. I can think of a scenario where, in the future, we want to use the code editor, but we don't want the button to be in this specific layout or anything.

I think we should remove the button from within this component, and make it up to the parent component to define a button for what to do with the value. I think the purpose of this component should be to just render the code editor and properly track its value. Not do any click handling.

This will be a bit of an involved refactor, so we can work on this together